pkg/cim/smb.go

```diff
@@ -0,0 +1,42 @@
+//go:build windows
+// +build windows
+
+package cim
+
+import (
+	"github.com/microsoft/wmi/pkg/base/query"
+	cim "github.com/microsoft/wmi/pkg/wmiinstance"
+)
+
+// QuerySmbGlobalMappingByRemotePath retrieves the SMB global mapping from its remote path.
+//
+// The equivalent WMI query is:
+//
+//	SELECT [selectors] FROM MSFT_SmbGlobalMapping
+//
+// Refer to https://pkg.go.dev/github.com/microsoft/wmi/server2019/root/microsoft/windows/smb#MSFT_SmbGlobalMapping
+// for the WMI class definition.
+func QuerySmbGlobalMappingByRemotePath(remotePath string) (*cim.WmiInstance, error) {
+	smbQuery := query.NewWmiQuery("MSFT_SmbGlobalMapping", "RemotePath", remotePath)
+	instances, err := QueryInstances(WMINamespaceSmb, smbQuery)
+	if err != nil {
+		return nil, err
```

Comment:
@andyzhangx I changed the return value to report error when not found to align with other methods in the same package.
`IsSmbMapped` method is also updated accordingly.

You may update your PR.
https://github.com/kubernetes-sigs/azurefile-csi-driver/pull/2532/files#diff-8887c406eb0dbea6fa1a8e31fdef2d34eb50a42119183907d7dd057f7d96a394R15